app/models/checkbox_facet.rb

```diff
@@ -0,0 +1,78 @@
+class CheckboxFacet < FilterableFacet
+  attr_writer :value
+
+  def checkboxes
+    @checkboxes ||= facet['checkboxes'].map { |checkbox| Checkbox.new(checkbox) }
+  end
+
+  def sentence_fragment
+    return nil if selected_checkboxes.empty?
+
+    {
+      'type' => "text",
+      'preposition' => preposition,
+      'values' => value_fragments,
+    }
+  end
+
+  def value
+    facet['value']
+  end
+
+  def checked?(checkbox)
+    @value && @value.include?(checkbox.value)
+  end
+
+private
+
+  def value_fragments
+    selected_checkboxes.map { |checkbox|
+      {
+        'label' => checkbox.label,
+        'parameter_key' => key,
+      }
+    }
+  end
+
+  def selected_checkboxes
+    return [] unless @value && @value.any?
+    checkboxes.select { |checkbox|
+      @value.include?(checkbox.value)
+    }
+  end
+end
+
+class Checkbox
```

Comment:
I thought about that, my thinking was that seeing `checkbox.rb` and `checkbox_facet.rb` in the same folder might be a little confusing and that this _might_ make it clearer that a `Checkbox` (sort of) belongs to a `CheckboxFacet`. If you feel strongly about it I'm happy to make the change

Reply:
Generally I disagree with multiple classes per file because it leads to larger files that are harder to skim read and understand the responsibility of the classes.

If the classes are very tied together, there's probably a better way to make the link between them clear while splitting them into separate files (maybe a module).

This also might be a sign that the classnames themselves are a bit too similar. More descriptive class + file names might help solve the confusion over multiple classes that do a similar thing but have different responsibilities.